### DISCUSSION TITLE
how to translate "Generate preview from file"

### DISCUSSION THREAD

**Discussion by punkyard** [te-online/timemanager#194] [General]
Hi,
I see this new button - but I could not translate it to French from the ln10 directory
Could I receive a guidance to complete translation? Thanks :)

![NC timemanager](https://github.com/user-attachments/assets/38f0ed6d-b6f7-4d30-8666-4f16b4669c7e)

**Comment by punkyard** [comment]
I see it only in [js/views/Import.svelte](https://github.com/te-online/timemanager/blob/e290de790556ab75d74d9885d7ff0eec64731074/js/views/Import.svelte#L226)

in French, it would be "Prévisualiser le fichier"

**Reply by te-online** [reply]
If a string is not in the translation files, you can add it 😊 

I've done that now ✅

**Reply by te-online** [reply]
Thanks for your contribution 🙌

**Comment by punkyard** [comment]
ha ha well i don't know how these translations files work, I didn't know I could just add a string myself and it will work ^^
so you did! I will have a look ;)

**Comment by punkyard** [comment]
well, in the end, i didn't find any new string to translate - i guess i didn't understand what you meant by "i've done that now"
I'll come back to see how i can translate more french - as i've seen other buttons in the plugin to translate

**Comment by te-online** [comment]
Hey 👋  That's totally fair 😅 

Every time in the code it uses the `translate()` or `t()` function, you can create a new string from what is given to this function. [See more details here](https://docs.nextcloud.com/server/stable/developer_manual/basics/front-end/l10n.html).

Unfortunately, I'm not always good at "extracting" all the keys, so that they actually are present in the .json and .js translation files. So it can be the case that some are missing 😳 

Let me know if you find any. 

> "i've done that now"

This just meant that I had used your provided translation and put it into the French file already 😊

**Comment by punkyard** [comment]
all right, then - i'm surely not able to participate to this
but if you can, let me know when it's done and I will surely translate them to french
take care till then

**Reply by te-online** [reply]
That's fine, thanks! 👍 

You're welcome to just drop any untranslated elements here and a small description of where you found them (or screenshot) and I will make sure to make them translatable 😊

**Comment by punkyard** [comment]
easy peasy

![Capture d’écran 2024-11-28 à 16 51 40](https://github.com/user-attachments/assets/94a9b204-f5d2-4591-8968-2cf271a41413)



see, all the buttons are not in the json, plus the other one I mentioned formerly 

👍

**Reply by te-online** [reply]
Thanks! I will take care of those and post here once it's done 😊

**Reply by te-online** [reply]
Hey 👋  I added the two missing strings to `main` 😊

**Comment by punkyard** [comment]
let me ask again: do you think there could be a way to link actions in time manager with tasks in Task app? with calDav or API?

**Reply by te-online** [reply]
There's definitely some way, but it needs to be coded, I'm afraid 😬 

You can always paste a link into the description of the time entry? This is how I used to do it in Toggl when I had to reference tasks.
